hs_composite_resource/models.py

```diff
@@ -577,6 +578,67 @@ def get_missing_file_type_metadata_info(self):
                                               'missing_elements': missing_elements})
         return metadata_missing_info
 
+    def get_data_services_urls(self):
+        """
+        Generates data services URLs for the resource.
+
+        If the resource contains any GeoFeature or GeoRaster content, and if it's public,
+        generate data service endpoints.
+        """
+
+        if self.raccess.public is True:
```

Comment:
All of this assumes the files are available on the geoserver and builds the url by convention.  It may make sense to set a flag on the resource object (within extra_data?) when the geoserver successfully retrieves the files.  This flag could be checked here so we have some assurance that the files are available on geoserver.  I'm open to other ideas and opinions on this.

Reply:
We don't have anywhere in HydroShare to store information about the data services right now. For now, we've decided to build the URLs assuming the services are available.